.github/workflows/build.yaml

```
@@ -0,0 +1,112 @@
+name: Build
+
+on:
+  workflow_call:
+    inputs:
+      mode:
+        required: true
+        type: string
+        default: snapshot
+        description: |
+          the mode to use. either `snapshot` or `release`. Will affect effective version, as well
+          as target-oci-registry.
+
+jobs:
+  prepare:
+    uses: gardener/cc-utils/.github/workflows/prepare.yaml@master
+    with:
+      mode: ${{ inputs.mode }}
+      version-commit-callback-action-path: ./.github/actions/prepare-release
+    permissions:
+      contents: read
+
+  sast-lint:
+    uses: gardener/cc-utils/.github/workflows/sastlint-ocm.yaml@master
+    permissions:
+      contents: read
+    with:
+      setup-git-identity: true
+      linter: gosec
+      run: |
+        set -euo pipefail
+        if [ '${{ inputs.mode }} ' == 'release' ]; then
```

Oof I missed this today during the review. 
```suggestion
        if [ '${{ inputs.mode }}' == 'release' ]; then
```